qubes/ext/core_features.py

```diff
@@ -62,3 +62,25 @@ def qubes_features_request(self, vm, event, untrusted_features):
             # if this is the first time qrexec was advertised, now can finish
             #  template setup
             yield from vm.fire_event_async('template-postinstall')
+
+    # pylint: disable=no-self-use
+    def set_servicevm_feature(self, subject):
+        if getattr(subject, 'provides_network', False):
+            subject.features['servicevm'] = 1
+        elif 'servicevm' in subject.features:
+            del subject.features['servicevm']
```

Comment:
I was deleting `servicevm` necessary? Later on this commit:

Came here from https://github.com/QubesOS/qubes-mgmt-salt-dom0-virtual-machines/pull/63

```python3
    def test_100_servicevm_feature(self):
        self.vm.provides_network = True
        self.ext.set_servicevm_feature(self.vm)
        self.assertEqual(self.features['servicevm'], 1)

        self.vm.provides_network = False
        self.ext.set_servicevm_feature(self.vm)
        self.assertNotIn('servicevm', self.features)
```

I don't think not having `provides_network` must enforce not having `servicevm`.